### DISCUSSION TITLE
Github Actions greetings.yaml addressing the issue opener directly by their username

### DISCUSSION THREAD

**Discussion by glenn-jocher** [community/community#25116] [Actions]
<p>I’m trying to build a yaml greeting for our repo:</p>
<p><a href="https://github.com/ultralytics/yolov3/blob/master/.github/workflows/greetings.yml" rel="nofollow noopener">https://github.com/ultralytics/yolov3/blob/master/.github/workflows/greetings.yml</a></p>
<p>I’d like to address the specific user in the message, for example <a class="mention" href="/glenn-jocher">@glenn-jocher</a> here:</p>
<p>“Hello <a class="mention" href="/glenn-jocher">@glenn-jocher</a>, thank you submitting a pull request!”</p>
<p>I can not find the correct syntax for this, how does this work? Thank you!</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/glenn-jocher">@glenn-jocher</a> ,</p>
<p>You can directly use the <strong><span class="mention">@mentioned</span></strong> in the message to send the message to the contributor.</p>
<p>For example:</p>
<pre><code>- name: First interaction
        uses: actions/first-interaction@v1
        with:
          repo-token: ${{ secrets.GITHUB_TOKEN }}
          issue-message: 'Hello @${{ github.actor }} , thank you for submitting an issue!'
          pr-message: 'Hello @${{ github.actor }} , thank you submitting a pull request!'
</code></pre>
<p>Generate message on the issue:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/31fc62efe0aaab6e3797d5bb12798e62b1d8ff27.png" data-download-href="/uploads/short-url/78c89Mfiep3MbacZf5Ph5a1CiLt.png?dl=1" title="example_issue.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/46307996/181088986-fb5bd8e1-3362-4a77-a34f-4d0da5d0afec.png" alt="example_issue.png" data-base62-sha1="78c89Mfiep3MbacZf5Ph5a1CiLt" width="690" height="300" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/31fc62efe0aaab6e3797d5bb12798e62b1d8ff27_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">example_issue.png</span><span class="informations">746×325 19.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Send message to my mail box:</p>
<p><img src="https://user-images.githubusercontent.com/46307996/181088987-fbab2432-37ea-4f54-ad41-cf2174b7843d.png" alt="example_email.png" data-base62-sha1="uPqEfizn7r2n8KgdwYZETigZ82o" width="497" height="243"></p>
<p>The property <strong>github.actor</strong> is used to get the login of the user that initiated the workflow run. More details, you can reference the docs about <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener"><strong>github context</strong></a>.</p>

**Comment by rohitthakur2590** [comment]
<p>It seems like</p>
<pre><code>uses: actions/first-interaction@v1
</code></pre>
<p>will greet only on the very first issue opened.<br>
what if we want to show the message on every new issue created. ? Any help would be appreciated.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/rohitthakur2590">@rohitthakur2590</a> ,</p>
<p>Actually, you can go to the <strong>GitHub marketplace</strong> to search the actions that can meet your request. On the marketplace, you can find all the GitHub’s official actions, and other actions published and shared by other users or services.</p>
<p>Here I search and find some actions about adding comments on issues,  maybe one of them is useful for you:</p>
<p><a href="https://github.com/marketplace?type=actions&amp;query=issue+comment" rel="nofollow noopener">https://github.com/marketplace?type=actions&amp;query=issue+comment</a></p>

**Comment by rohitthakur2590** [comment]
<p>Thanks <a class="mention" href="/brightran">@brightran</a>  Hope I can get something out of this.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/rohitthakur2590">@rohitthakur2590</a> you’re welcome.</p>

**Comment by Prateek-python** [comment]
You can use this:




name: Auto Comment
on: [issues, pull_request]
jobs:
  run:
    runs-on: ubuntu-latest
    steps:
      - uses: wow-actions/auto-comment@v1
        with:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          issuesOpened: |
            👋 @{{ author }}
            Thank you for raising an issue. We will will investigate into the matter and get back to you as soon as possible.
            Please make sure you have given us as much context as possible.

          pullRequestOpened: |
            👋 @{{ author }}
            Thank you for raising your pull request.
            Please make sure you have followed our contributing guidelines. We will review it as soon as possible
